lisa/analysis/cpus.py

```diff
@@ -66,6 +66,96 @@ def df_context_switches(self):
 
         return ctx_sw_df
 
+    @TraceAnalysisBase.cache
+    @requires_events('sched_switch')
+    def df_states(self):
```

Comment:
Maybe we could have a name that reflects the kind of state we are talking about ?

Currently it's idle/busy, as defined from `sched_switch` only. It does not account for e.g. interrupts, kernel work happening before sched_switch and does not deal with cpuidle states. Something like `df_task_active_states()` would match `tasks.tasks_states()` `ACTIVE` state.

Should we leave the door opened to expanding the number of states in the future (e.g. someone could add INTERRUPT_BUSY for interrupts) or do we want to solidify this specific definition ? Either way, it would need to be stated in the docstring.